pkgs/test_core/lib/src/runner/engine.dart

```diff
@@ -265,35 +256,32 @@ class Engine {
       _onSuiteAddedController.add(suite);
 
       _group.add(() async {
-        var loadResource = await _loadPool.request();
-
+        var resource = await _runPool.request();
         LiveSuiteController controller;
-        if (suite is LoadSuite) {
-          await _onUnpaused;
-          controller = await _addLoadSuite(suite);
-          if (controller == null) {
-            loadResource.release();
-            return;
+        try {
```

Comment:
Why do we now need a try catch clause?

Reply:
This is just for the finally to ensure that we release the resource from the pool

Reply:
(we return a couple places from this function, and want to make sure we release it even if there is an error as well)